### PR TITLE
Move to GitHub actions

### PR DESCRIPTION
**Pull request by Chrizpy**

### CODE REVIEW

**Comment by Chrizpy** [comment]
[Sprint88: Migrate public repositories to travis-ci.com](https://trello.com/c/j1LLmTJh/2623-sprint88-migrate-public-repositories-to-travis-cicom)

**Comment by walro** [comment]
I think it makes the most sense that the removal of `.travis.yml` comes last, there is no CI of the commits in between otherwise.

Also, squash all three commits regarding GitHub actions to tidy of the history.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+
+name: CI
+
+on: push
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
```
Comment (walro) [inline .github/workflows/ci.yml:8]: Ideally we run on Bionic (18.04), is that possible?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+    strategy:
+      matrix:
+        ruby: [2.7.1, head]
```
Comment (walro) [inline .github/workflows/ci.yml:12]: in the Travis case we allow failures on head, I don't see that ported here

**Comment by Chrizpy** [comment]
> I think it makes the most sense that the removal of `.travis.yml` comes last, there is no CI of the commits in between otherwise.

Yeah that makes sense, I did not think about that.

> Also, squash all three commits regarding GitHub actions to tidy up the history.

Sure thing, I like squashing at the end, just so that whenever someone comes over they can see how I have stumbled around. :)

File: .github/workflows/ci.yml (hunk 2 above)
Comment (Chrizpy) [reply .github/workflows/ci.yml:12]: Doesn't seem like there is an official feature of this yet in GitHub actions. Many people request this though, seems like many more are migrating from Travis like we are doing. https://github.com/actions/toolkit/issues/399 This thread has been ongoing for a while now and is updated very frequently.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Chrizpy) [reply .github/workflows/ci.yml]: Shouldn't be a problem: 4be14ec8a722cc10ff36c8fc91a29926750d477a
:)

File: .github/workflows/ci.yml (hunk 2 above)
Comment (walro) [reply .github/workflows/ci.yml:12]: Ah, too bad. Ah well, if we start getting failures we can just remove `head` for the time being.

Comment (walro) [reply .github/workflows/ci.yml:12]: (I subscribed to the issue)

**Comment by walro** [comment]
> Sure thing, I like squashing at the end,

Yeah, that's what I meant :)

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+
```
Comment (walro) [inline .github/workflows/ci.yml:1]: haven't thought of this before, but is it common that the workflow files start with newline?

(It's common that they start with `---`, but I guess that's optional)

Comment (Chrizpy) [reply .github/workflows/ci.yml:1]: No it's actually a typo haha. I do think I remember seeing `---` in Ansible stuff though? I'll have a looksie.

Comment (Chrizpy) [reply .github/workflows/ci.yml:1]: Seems to be some sort of `.yml` practise. 
When many files are read into a stream, the three dashes are meant to be the separators.
https://stackoverflow.com/questions/50788277/why-3-dashes-hyphen-in-yaml-file

Comment (walro) [reply .github/workflows/ci.yml:1]: yeah, but does GitHub workflow files bother about that (in general)

Comment (Chrizpy) [reply .github/workflows/ci.yml:1]: I don't find it in the GitHub workflows guides though.

Comment (Chrizpy) [reply .github/workflows/ci.yml]: 0b1b683bc82d48d92dce35b44b9c95a478d8d7ed

Comment (walro) [reply .github/workflows/ci.yml]: Looking at https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#about-yaml-syntax-for-workflows I would say that triple dash is not really used within the GitHub Actions community.

**Review by walro — APPROVED** [review]
LGTM!

(I don't know if there's any cleanup needed over at Travis CI)

**Pull request merged**